{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4WjCUKhOZbJk",
        "outputId": "7b7e0ea4-435e-40e4-8ec3-e61bea12a093"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting macs2\n",
            "  Downloading MACS2-2.2.9.1.tar.gz (2.0 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m2.0/2.0 MB\u001b[0m \u001b[31m11.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy>=1.19 in /usr/local/lib/python3.11/dist-packages (from macs2) (1.26.4)\n",
            "Collecting Cython~=0.29 (from macs2)\n",
            "  Using cached Cython-0.29.37-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl.metadata (3.1 kB)\n",
            "Using cached Cython-0.29.37-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)\n",
            "Building wheels for collected packages: macs2\n",
            "  Building wheel for macs2 (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for macs2: filename=macs2-2.2.9.1-cp311-cp311-linux_x86_64.whl size=1828750 sha256=fc059df1c419e223a0e7da79131d968c3d3bae1ec259f7b204f298747fb13eeb\n",
            "  Stored in directory: /root/.cache/pip/wheels/fa/c0/cd/e20c4ce750b1247dc1432a11eaca830948bab4e5326997636d\n",
            "Successfully built macs2\n",
            "Installing collected packages: Cython, macs2\n",
            "  Attempting uninstall: Cython\n",
            "    Found existing installation: Cython 3.0.12\n",
            "    Uninstalling Cython-3.0.12:\n",
            "      Successfully uninstalled Cython-3.0.12\n",
            "Successfully installed Cython-0.29.37 macs2-2.2.9.1\n"
          ]
        }
      ],
      "source": [
        "!pip install macs2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -q condacolab"
      ],
      "metadata": {
        "id": "lSxRAltzZngW"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "import condacolab"
      ],
      "metadata": {
        "id": "uRvGAMXkZs1R"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "condacolab.install()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bX22IdT5ZvhX",
        "outputId": "94d19746-ddf4-49f0-85cb-3e50260c870f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "âœ¨ðŸ°âœ¨ Everything looks OK!\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda seqtk"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9Np_Yka9ao-q",
        "outputId": "5f6bdd04-6e50-40c1-a801-acbf068bcbd6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - seqtk\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    ca-certificates-2025.1.31  |       hbcca054_0         154 KB  conda-forge\n",
            "    certifi-2025.1.31          |     pyhd8ed1ab_0         159 KB  conda-forge\n",
            "    conda-24.11.3              |  py311h38be061_0         1.1 MB  conda-forge\n",
            "    openssl-3.4.1              |       h7b32b05_0         2.8 MB  conda-forge\n",
            "    seqtk-1.4                  |       h577a1d6_3          43 KB  bioconda\n",
            "    zlib-1.3.1                 |       hb9d3cd8_2          90 KB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:         4.4 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  seqtk              bioconda/linux-64::seqtk-1.4-h577a1d6_3 \n",
            "  zlib               conda-forge/linux-64::zlib-1.3.1-hb9d3cd8_2 \n",
            "\n",
            "The following packages will be UPDATED:\n",
            "\n",
            "  ca-certificates                     2024.12.14-hbcca054_0 --> 2025.1.31-hbcca054_0 \n",
            "  certifi                           2024.12.14-pyhd8ed1ab_0 --> 2025.1.31-pyhd8ed1ab_0 \n",
            "  conda                             24.11.2-py311h38be061_1 --> 24.11.3-py311h38be061_0 \n",
            "  openssl                                  3.4.0-h7b32b05_1 --> 3.4.1-h7b32b05_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "openssl-3.4.1        | 2.8 MB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "conda-24.11.3        | 1.1 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "certifi-2025.1.31    | 159 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "ca-certificates-2025 | 154 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.3.1           | 90 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "seqtk-1.4            | 43 KB     | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "certifi-2025.1.31    | 159 KB    | : 100% 1.0/1 [00:00<00:00, 12.78it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.4.1        | 2.8 MB    | :   1% 0.0055741049077571376/1 [00:00<00:19, 19.16s/it]\n",
            "\n",
            "\n",
            "ca-certificates-2025 | 154 KB    | :  10% 0.10360178065560502/1 [00:00<00:00,  1.10s/it]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "seqtk-1.4            | 43 KB     | :  37% 0.3717637448662386/1 [00:00<00:00,  3.33it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "ca-certificates-2025 | 154 KB    | : 100% 1.0/1 [00:00<00:00,  1.10s/it]                \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "seqtk-1.4            | 43 KB     | : 100% 1.0/1 [00:00<00:00,  3.33it/s]               \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "conda-24.11.3        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  6.67it/s]\u001b[A\n",
            "conda-24.11.3        | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  6.67it/s]\u001b[A\n",
            "\n",
            "certifi-2025.1.31    | 159 KB    | : 100% 1.0/1 [00:00<00:00,  6.40it/s]\u001b[A\u001b[A\n",
            "\n",
            "certifi-2025.1.31    | 159 KB    | : 100% 1.0/1 [00:00<00:00,  6.40it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.3.1           | 90 KB     | : 100% 1.0/1 [00:00<00:00,  6.17it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "zlib-1.3.1           | 90 KB     | : 100% 1.0/1 [00:00<00:00,  6.17it/s]\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "ca-certificates-2025 | 154 KB    | : 100% 1.0/1 [00:00<00:00,  1.10s/it]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "seqtk-1.4            | 43 KB     | : 100% 1.0/1 [00:00<00:00,  4.98it/s]\u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "openssl-3.4.1        | 2.8 MB    | : 100% 1.0/1 [00:00<00:00,  5.01it/s]\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\b/ \b\b- \b\bdone\n",
            "Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda trimmomatic"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "hZO96NXPasm9",
        "outputId": "a0e0d4c3-4df4-42b0-e0a1-a392f5616254"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "\n",
            "==> WARNING: A newer version of conda exists. <==\n",
            "    current version: 24.11.3\n",
            "    latest version: 25.1.1\n",
            "\n",
            "Please update conda by running\n",
            "\n",
            "    $ conda update -n base -c conda-forge conda\n",
            "\n",
            "\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - trimmomatic\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    openjdk-11.0.1             |    h516909a_1016       175.5 MB  conda-forge\n",
            "    trimmomatic-0.39           |       hdfd78af_2         144 KB  bioconda\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:       175.7 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  openjdk            conda-forge/linux-64::openjdk-11.0.1-h516909a_1016 \n",
            "  trimmomatic        bioconda/noarch::trimmomatic-0.39-hdfd78af_2 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages:\n",
            "openjdk-11.0.1       | 175.5 MB  | :   0% 0/1 [00:00<?, ?it/s]\n",
            "trimmomatic-0.39     | 144 KB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "openjdk-11.0.1       | 175.5 MB  | :   2% 0.015579077917781254/1 [00:00<00:06,  6.42s/it]\n",
            "trimmomatic-0.39     | 144 KB    | : 100% 1.0/1 [00:00<00:00,  8.12it/s]\u001b[A\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "Preparing transaction: - \b\bdone\n",
            "Verifying transaction: | \b\bdone\n",
            "Executing transaction: - \b\bdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install intervene"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "O1A-Sm_0azoz",
        "outputId": "87262ae8-19e1-4082-93e0-0bcf263182f3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting intervene\n",
            "  Downloading intervene-0.6.5.tar.gz (5.5 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m5.5/5.5 MB\u001b[0m \u001b[31m53.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting pybedtools (from intervene)\n",
            "  Downloading pybedtools-0.11.0.tar.gz (12.5 MB)\n",
            "\u001b[2K     \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m12.5/12.5 MB\u001b[0m \u001b[31m128.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "Collecting matplotlib (from intervene)\n",
            "  Downloading matplotlib-3.10.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (11 kB)\n",
            "Collecting pandas (from intervene)\n",
            "  Downloading pandas-2.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (89 kB)\n",
            "Collecting numpy (from intervene)\n",
            "  Using cached numpy-2.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (62 kB)\n",
            "Collecting scipy (from intervene)\n",
            "  Downloading scipy-1.15.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (61 kB)\n",
            "Collecting seaborn (from intervene)\n",
            "  Downloading seaborn-0.13.2-py3-none-any.whl.metadata (5.4 kB)\n",
            "Collecting contourpy>=1.0.1 (from matplotlib->intervene)\n",
            "  Downloading contourpy-1.3.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.4 kB)\n",
            "Collecting cycler>=0.10 (from matplotlib->intervene)\n",
            "  Downloading cycler-0.12.1-py3-none-any.whl.metadata (3.8 kB)\n",
            "Collecting fonttools>=4.22.0 (from matplotlib->intervene)\n",
            "  Downloading fonttools-4.56.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (101 kB)\n",
            "Collecting kiwisolver>=1.3.1 (from matplotlib->intervene)\n",
            "  Downloading kiwisolver-1.4.8-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.2 kB)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.11/site-packages (from matplotlib->intervene) (24.2)\n",
            "Collecting pillow>=8 (from matplotlib->intervene)\n",
            "  Downloading pillow-11.1.0-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (9.1 kB)\n",
            "Collecting pyparsing>=2.3.1 (from matplotlib->intervene)\n",
            "  Downloading pyparsing-3.2.1-py3-none-any.whl.metadata (5.0 kB)\n",
            "Collecting python-dateutil>=2.7 (from matplotlib->intervene)\n",
            "  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)\n",
            "Collecting pytz>=2020.1 (from pandas->intervene)\n",
            "  Downloading pytz-2025.1-py2.py3-none-any.whl.metadata (22 kB)\n",
            "Collecting tzdata>=2022.7 (from pandas->intervene)\n",
            "  Downloading tzdata-2025.1-py2.py3-none-any.whl.metadata (1.4 kB)\n",
            "Collecting pysam (from pybedtools->intervene)\n",
            "  Downloading pysam-0.23.0-cp311-cp311-manylinux_2_28_x86_64.whl.metadata (1.6 kB)\n",
            "Collecting six>=1.5 (from python-dateutil>=2.7->matplotlib->intervene)\n",
            "  Downloading six-1.17.0-py2.py3-none-any.whl.metadata (1.7 kB)\n",
            "Downloading matplotlib-3.10.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (8.6 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m8.6/8.6 MB\u001b[0m \u001b[31m122.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hUsing cached numpy-2.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (16.4 MB)\n",
            "Downloading pandas-2.2.3-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (13.1 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m13.1/13.1 MB\u001b[0m \u001b[31m119.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading scipy-1.15.2-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (37.6 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m37.6/37.6 MB\u001b[0m \u001b[31m45.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading seaborn-0.13.2-py3-none-any.whl (294 kB)\n",
            "Downloading contourpy-1.3.1-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (326 kB)\n",
            "Downloading cycler-0.12.1-py3-none-any.whl (8.3 kB)\n",
            "Downloading fonttools-4.56.0-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.9 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m4.9/4.9 MB\u001b[0m \u001b[31m87.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading kiwisolver-1.4.8-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.4 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m1.4/1.4 MB\u001b[0m \u001b[31m39.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pillow-11.1.0-cp311-cp311-manylinux_2_28_x86_64.whl (4.5 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m4.5/4.5 MB\u001b[0m \u001b[31m81.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading pyparsing-3.2.1-py3-none-any.whl (107 kB)\n",
            "Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)\n",
            "Downloading pytz-2025.1-py2.py3-none-any.whl (507 kB)\n",
            "Downloading tzdata-2025.1-py2.py3-none-any.whl (346 kB)\n",
            "Downloading pysam-0.23.0-cp311-cp311-manylinux_2_28_x86_64.whl (26.0 MB)\n",
            "\u001b[2K   \u001b[90mâ”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”â”\u001b[0m \u001b[32m26.0/26.0 MB\u001b[0m \u001b[31m91.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hDownloading six-1.17.0-py2.py3-none-any.whl (11 kB)\n",
            "Building wheels for collected packages: intervene, pybedtools\n",
            "  Building wheel for intervene (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for intervene: filename=intervene-0.6.5-py3-none-any.whl size=5546123 sha256=5dcc2ef970059d36fc311aa20af9de77cd66c1498a06ff9eec5aa9467ac1f15e\n",
            "  Stored in directory: /root/.cache/pip/wheels/3f/ed/5c/4ad38469946289c3545900cdac0d39aca0bd7890ab5f931f7b\n",
            "  Building wheel for pybedtools (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for pybedtools: filename=pybedtools-0.11.0-cp311-cp311-linux_x86_64.whl size=12749692 sha256=481575a57fe1fd141e82e2d230272df62d5cfd22f2dd4c62b6b2d5c96748fb9d\n",
            "  Stored in directory: /root/.cache/pip/wheels/29/a9/64/b72ced6183351e9fb889c0fc86eda39fab286a4b8838443314\n",
            "Successfully built intervene pybedtools\n",
            "Installing collected packages: pytz, tzdata, six, pysam, pyparsing, pillow, numpy, kiwisolver, fonttools, cycler, scipy, python-dateutil, pybedtools, contourpy, pandas, matplotlib, seaborn, intervene\n",
            "Successfully installed contourpy-1.3.1 cycler-0.12.1 fonttools-4.56.0 intervene-0.6.5 kiwisolver-1.4.8 matplotlib-3.10.1 numpy-2.2.3 pandas-2.2.3 pillow-11.1.0 pybedtools-0.11.0 pyparsing-3.2.1 pysam-0.23.0 python-dateutil-2.9.0.post0 pytz-2025.1 scipy-1.15.2 seaborn-0.13.2 six-1.17.0 tzdata-2025.1\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "application/vnd.colab-display-data+json": {
              "pip_warning": {
                "packages": [
                  "PIL",
                  "cycler",
                  "kiwisolver",
                  "six"
                ]
              },
              "id": "f4232dae79ec4de28fa8dae8468f024c"
            }
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install bedtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8D3UL8L7a321",
        "outputId": "4e8f95a7-eace-4e66-c3fb-c8f491cdfd01"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 29 not upgraded.\n",
            "Need to get 563 kB of archives.\n",
            "After this operation, 1,548 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 bedtools amd64 2.30.0+dfsg-2ubuntu0.1 [563 kB]\n",
            "Fetched 563 kB in 2s (313 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 124947 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.30.0+dfsg-2ubuntu0.1_amd64.deb ...\n",
            "Unpacking bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n",
            "Setting up bedtools (2.30.0+dfsg-2ubuntu0.1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get install bowtie2 samtools"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zktWk67HbFg5",
        "outputId": "1319226e-6b14-41b6-dfbc-2b31901e0208"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "bowtie2 is already the newest version (2.4.4-1).\n",
            "samtools is already the newest version (1.13-4).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 29 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!conda install -c bioconda bowtie2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "doLfJTtUbF_T",
        "outputId": "a337ce64-cf21-494a-8ea0-2512bc8c1662"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Channels:\n",
            " - bioconda\n",
            " - conda-forge\n",
            "Platform: linux-64\n",
            "Collecting package metadata (repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\bdone\n",
            "\n",
            "\n",
            "==> WARNING: A newer version of conda exists. <==\n",
            "    current version: 24.11.3\n",
            "    latest version: 25.1.1\n",
            "\n",
            "Please update conda by running\n",
            "\n",
            "    $ conda update -n base -c conda-forge conda\n",
            "\n",
            "\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ° FastQC"
      ],
      "metadata": {
        "id": "u2qf1571bIz7"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!java -version"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9XBB00EPbKrq",
        "outputId": "b07e97e2-3c24-44bc-8f6c-90e8632a404b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"11.0.1\" 2018-10-16 LTS\n",
            "OpenJDK Runtime Environment Zulu11.2+3 (build 11.0.1+13-LTS)\n",
            "OpenJDK 64-Bit Server VM Zulu11.2+3 (build 11.0.1+13-LTS, mixed mode)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3_8BoXoCbMeA",
        "outputId": "473690bc-b698-4d82-be52-580da02b17ae"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-03-08 11:05:01--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: â€˜fastqc_v0.11.9.zipâ€™\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  4.30MB/s    in 2.3s    \n",
            "\n",
            "2025-03-08 11:05:05 (4.30 MB/s) - â€˜fastqc_v0.11.9.zipâ€™ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JmS-kT6QbOZJ",
        "outputId": "9fcd6d6a-be3c-4fc2-e210-a82d08103d1c"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!chmod a+x FastQC/fastqc"
      ],
      "metadata": {
        "id": "5TKEKlbIbhaZ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc --help"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "-2xp-qi8bk-u",
        "outputId": "51f352ba-5166-449b-8ab0-dd155e21b93a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Ð’Ñ‹Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°Ð½Ð¸Ðµ"
      ],
      "metadata": {
        "id": "XKCYZLVybl5m"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "from google.colab import drive\n",
        "drive.mount('/content/drive')"
      ],
      "metadata": {
        "id": "6hglWv3aboC0",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "51c9d227-4d7f-4405-ffc9-d202e2020630"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Mounted at /content/drive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%cd /content/drive/MyDrive"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DcfUv66TmzhE",
        "outputId": "58f91d72-70f0-452b-d84e-9622e5bb6dc7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/drive/MyDrive\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Ñ€Ð°ÑÐ¿Ð°ÐºÐ¾Ð²ÐºÐ° Ð°Ñ…Ñ€Ð¸Ð²Ð°\n",
        "!gzip -d ENCFF137PVL.fastq.gz ENCFF171DUV.fastq.gz ENCFF176RJQ.fastq.gz"
      ],
      "metadata": {
        "id": "v9shFgXLm9z2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!/content/FastQC/fastqc ENCFF137PVL.fastq\n",
        "!/content/FastQC/fastqc ENCFF171DUV.fastq\n",
        "!/content/FastQC/fastqc ENCFF176RJQ.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7e_poxjcnnfb",
        "outputId": "a8f68711-e1d4-4eb7-b264-fc0c4900dec6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF137PVL.fastq\n",
            "Approx 5% complete for ENCFF137PVL.fastq\n",
            "Approx 10% complete for ENCFF137PVL.fastq\n",
            "Approx 15% complete for ENCFF137PVL.fastq\n",
            "Approx 20% complete for ENCFF137PVL.fastq\n",
            "Approx 25% complete for ENCFF137PVL.fastq\n",
            "Approx 30% complete for ENCFF137PVL.fastq\n",
            "Approx 35% complete for ENCFF137PVL.fastq\n",
            "Approx 40% complete for ENCFF137PVL.fastq\n",
            "Approx 45% complete for ENCFF137PVL.fastq\n",
            "Approx 50% complete for ENCFF137PVL.fastq\n",
            "Approx 55% complete for ENCFF137PVL.fastq\n",
            "Approx 60% complete for ENCFF137PVL.fastq\n",
            "Approx 65% complete for ENCFF137PVL.fastq\n",
            "Approx 70% complete for ENCFF137PVL.fastq\n",
            "Approx 75% complete for ENCFF137PVL.fastq\n",
            "Approx 80% complete for ENCFF137PVL.fastq\n",
            "Approx 85% complete for ENCFF137PVL.fastq\n",
            "Approx 90% complete for ENCFF137PVL.fastq\n",
            "Approx 95% complete for ENCFF137PVL.fastq\n",
            "Analysis complete for ENCFF137PVL.fastq\n",
            "Started analysis of ENCFF171DUV.fastq\n",
            "Approx 5% complete for ENCFF171DUV.fastq\n",
            "Approx 10% complete for ENCFF171DUV.fastq\n",
            "Approx 15% complete for ENCFF171DUV.fastq\n",
            "Approx 20% complete for ENCFF171DUV.fastq\n",
            "Approx 25% complete for ENCFF171DUV.fastq\n",
            "Approx 30% complete for ENCFF171DUV.fastq\n",
            "Approx 35% complete for ENCFF171DUV.fastq\n",
            "Approx 40% complete for ENCFF171DUV.fastq\n",
            "Approx 45% complete for ENCFF171DUV.fastq\n",
            "Approx 50% complete for ENCFF171DUV.fastq\n",
            "Approx 55% complete for ENCFF171DUV.fastq\n",
            "Approx 60% complete for ENCFF171DUV.fastq\n",
            "Approx 65% complete for ENCFF171DUV.fastq\n",
            "Approx 70% complete for ENCFF171DUV.fastq\n",
            "Approx 75% complete for ENCFF171DUV.fastq\n",
            "Approx 80% complete for ENCFF171DUV.fastq\n",
            "Approx 85% complete for ENCFF171DUV.fastq\n",
            "Approx 90% complete for ENCFF171DUV.fastq\n",
            "Approx 95% complete for ENCFF171DUV.fastq\n",
            "Analysis complete for ENCFF171DUV.fastq\n",
            "Started analysis of ENCFF176RJQ.fastq\n",
            "Approx 5% complete for ENCFF176RJQ.fastq\n",
            "Approx 10% complete for ENCFF176RJQ.fastq\n",
            "Approx 15% complete for ENCFF176RJQ.fastq\n",
            "Approx 20% complete for ENCFF176RJQ.fastq\n",
            "Approx 25% complete for ENCFF176RJQ.fastq\n",
            "Approx 30% complete for ENCFF176RJQ.fastq\n",
            "Approx 35% complete for ENCFF176RJQ.fastq\n",
            "Approx 40% complete for ENCFF176RJQ.fastq\n",
            "Approx 45% complete for ENCFF176RJQ.fastq\n",
            "Approx 50% complete for ENCFF176RJQ.fastq\n",
            "Approx 55% complete for ENCFF176RJQ.fastq\n",
            "Approx 60% complete for ENCFF176RJQ.fastq\n",
            "Approx 65% complete for ENCFF176RJQ.fastq\n",
            "Approx 70% complete for ENCFF176RJQ.fastq\n",
            "Approx 75% complete for ENCFF176RJQ.fastq\n",
            "Approx 80% complete for ENCFF176RJQ.fastq\n",
            "Approx 85% complete for ENCFF176RJQ.fastq\n",
            "Approx 90% complete for ENCFF176RJQ.fastq\n",
            "Approx 95% complete for ENCFF176RJQ.fastq\n",
            "Analysis complete for ENCFF176RJQ.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF137PVL.fastq ENCFF137PVL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xP7D90CYonDg",
        "outputId": "46cac69c-3198-4f89-e4d5-4948c26d7c7b"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF137PVL.fastq ENCFF137PVL_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/drive/MyDrive/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 8874861 Surviving: 4005866 (45.14%) Dropped: 4868995 (54.86%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF171DUV.fastq ENCFF171DUV_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "X15tvcAdrvfU",
        "outputId": "470043d2-9ead-4441-c76f-ec1e5da4a270"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF171DUV.fastq ENCFF171DUV_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/drive/MyDrive/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 19825041 Surviving: 12142545 (61.25%) Dropped: 7682496 (38.75%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF176RJQ.fastq ENCFF176RJQ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JTNgwHHUsVK3",
        "outputId": "62fa5437-c711-4544-e1a0-fc75eb997375"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF176RJQ.fastq ENCFF176RJQ_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/drive/MyDrive/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:219)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 9357767 Surviving: 5229995 (55.89%) Dropped: 4127772 (44.11%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!/content/FastQC/fastqc /content/drive/MyDrive/ENCFF137PVL_trimmed.fastq\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YK8WmNEZvTj_",
        "outputId": "6a83ddd9-d3ef-4a45-9d5b-503bcb592c80"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF137PVL_trimmed.fastq\n",
            "Approx 5% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 10% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 15% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 20% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 25% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 30% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 35% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 40% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 45% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 50% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 55% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 60% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 65% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 70% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 75% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 80% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 85% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 90% complete for ENCFF137PVL_trimmed.fastq\n",
            "Approx 95% complete for ENCFF137PVL_trimmed.fastq\n",
            "Analysis complete for ENCFF137PVL_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!/content/FastQC/fastqc /content/drive/MyDrive/ENCFF176RJQ_trimmed.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DkWJDOk-wHqV",
        "outputId": "cc15682b-7b07-41b8-e3a1-c1bc142d5c36"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF176RJQ_trimmed.fastq\n",
            "Approx 5% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 10% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 15% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 20% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 25% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 30% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 35% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 40% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 45% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 50% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 55% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 60% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 65% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 70% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 75% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 80% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 85% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 90% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Approx 95% complete for ENCFF176RJQ_trimmed.fastq\n",
            "Analysis complete for ENCFF176RJQ_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!/content/FastQC/fastqc /content/drive/MyDrive/ENCFF171DUV_trimmed.fastq"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QAwWsAZfwTXL",
        "outputId": "6785a240-a8dd-47f0-d629-a970712b543f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF171DUV_trimmed.fastq\n",
            "Approx 5% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 10% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 15% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 20% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 25% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 30% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 35% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 40% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 45% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 50% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 55% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 60% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 65% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 70% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 75% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 80% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 85% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 90% complete for ENCFF171DUV_trimmed.fastq\n",
            "Approx 95% complete for ENCFF171DUV_trimmed.fastq\n",
            "Analysis complete for ENCFF171DUV_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## 4. Ð’Ñ‹Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°Ð½Ð¸Ðµ Ð½Ð° Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ\n",
        "\n",
        "Ð¡Ð»ÐµÐ´ÑƒÐµÑ‚ Ð²Ñ‹Ð±Ñ€Ð°Ñ‚ÑŒ Ð¾Ð´Ð½Ñƒ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ, Ð¿Ð¾Ñ‚Ð¾Ð¼Ñƒ Ñ‡Ñ‚Ð¾ Ñ€ÐµÑÑƒÑ€ÑÑ‹ google colab Ð¾Ð³Ñ€Ð°Ð½Ð¸Ñ‡ÐµÐ½Ñ‹. Ð’ ÑÐ»ÑƒÑ‡Ð°Ðµ Ð¿Ñ€Ð¾Ð±Ð»ÐµÐ¼, Ñ‚Ð°ÐºÐ¸Ñ… ÐºÐ°Ðº Segmentation fault, error 139, ÑÐ»ÐµÐ´ÑƒÐµÑ‚ Ð²Ñ‹Ð±Ñ€Ð°Ñ‚ÑŒ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ Ð¼ÐµÐ½ÑŒÑˆÐµÐ³Ð¾ Ñ€Ð°Ð·Ð¼ÐµÑ€Ð°.\n",
        "\n",
        "ÐŸÐ¾ÑÐ»ÐµÐ´Ð¾Ð²Ð°Ñ‚ÐµÐ»ÑŒÐ½Ð¾ÑÑ‚ÑŒ Ð½ÑƒÐºÐ»ÐµÐ¾Ñ‚Ð¸Ð´Ð¾Ð² Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñ‹ Ð¼Ð¾Ð¶Ð½Ð¾ ÑÐºÐ°Ñ‡Ð°Ñ‚ÑŒ Ð¿Ð¾ Ð°Ð´Ñ€ÐµÑÑƒ https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ],
      "metadata": {
        "id": "Wj-HUIICwh9t"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
        "!gzip -d chr14.fa.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_9sDcGHywe63",
        "outputId": "0330cf2f-412c-4899-e0e1-53f3f35da63b"
      },
      "execution_count": null,
      "outputs": [
        {
          "metadata": {
            "tags": null
          },
          "name": "stdout",
          "output_type": "stream",
          "text": [
            "--2025-03-08 19:17:34--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr14.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 29295890 (28M) [application/x-gzip]\n",
            "Saving to: â€˜chr14.fa.gzâ€™\n",
            "\n",
            "chr14.fa.gz         100%[===================>]  27.94M  13.6MB/s    in 2.0s    \n",
            "\n",
            "2025-03-08 19:17:37 (13.6 MB/s) - â€˜chr14.fa.gzâ€™ saved [29295890/29295890]\n",
            "\n",
            "gzip: chr14.fa already exists; do you wish to overwrite (y or n)? "
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2-build chr14.fa chromosome_index"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "9WBcN2SHHmiw",
        "outputId": "f3885afd-5395-423e-9981-85bb781893e3"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr1.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:04\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:04\n",
            "bmax according to bmaxDivN setting: 57620253\n",
            "Using parameters --bmax 43215190 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 43215190 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:09\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:03\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:03\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 3.29259e+07 (target: 43215189)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 7\n",
            "  Reserving size (43215190) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 41559493 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:36\n",
            "Returning block of 41559494 for bucket 1\n",
            "Getting block 2 of 7\n",
            "  Reserving size (43215190) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 36821901 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:32\n",
            "Returning block of 36821902 for bucket 2\n",
            "Getting block 3 of 7\n",
            "  Reserving size (43215190) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 41919124 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:38\n",
            "Returning block of 41919125 for bucket 3\n",
            "Getting block 4 of 7\n",
            "  Reserving size (43215190) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 24681605 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:20\n",
            "Returning block of 24681606 for bucket 4\n",
            "Getting block 5 of 7\n",
            "  Reserving size (43215190) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 41918922 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:41\n",
            "Returning block of 41918923 for bucket 5\n",
            "Getting block 6 of 7\n",
            "  Reserving size (43215190) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 8244387 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:07\n",
            "Returning block of 8244388 for bucket 6\n",
            "Getting block 7 of 7\n",
            "  Reserving size (43215190) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 35335574 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:30\n",
            "Returning block of 35335575 for bucket 7\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 67070277\n",
            "fchr[G]: 115125320\n",
            "fchr[T]: 163236848\n",
            "fchr[$]: 230481012\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 81023470 bytes to primary EBWT file: chromosome_index.1.bt2.tmp\n",
            "Wrote 57620260 bytes to secondary EBWT file: chromosome_index.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 230481012\n",
            "    bwtLen: 230481013\n",
            "    sz: 57620253\n",
            "    bwtSz: 57620254\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 14405064\n",
            "    offsSz: 57620256\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 1200422\n",
            "    numLines: 1200422\n",
            "    ebwtTotLen: 76827008\n",
            "    ebwtTotSz: 76827008\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:05:51\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:02\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:02\n",
            "  Time to reverse reference sequence: 00:00:01\n",
            "bmax according to bmaxDivN setting: 57620253\n",
            "Using parameters --bmax 43215190 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 43215190 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:08\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:02\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:03\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 2, merged 6; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 1; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 1; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 1; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 2.88101e+07 (target: 43215189)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 8\n",
            "  Reserving size (43215190) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 28292839 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:24\n",
            "Returning block of 28292840 for bucket 1\n",
            "Getting block 2 of 8\n",
            "  Reserving size (43215190) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 39729933 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:38\n",
            "Returning block of 39729934 for bucket 2\n",
            "Getting block 3 of 8\n",
            "  Reserving size (43215190) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 14119472 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:14\n",
            "Returning block of 14119473 for bucket 3\n",
            "Getting block 4 of 8\n",
            "  Reserving size (43215190) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 41568163 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:37\n",
            "Returning block of 41568164 for bucket 4\n",
            "Getting block 5 of 8\n",
            "  Reserving size (43215190) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 37733383 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:33\n",
            "Returning block of 37733384 for bucket 5\n",
            "Getting block 6 of 8\n",
            "  Reserving size (43215190) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 23633964 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:20\n",
            "Returning block of 23633965 for bucket 6\n",
            "Getting block 7 of 8\n",
            "  Reserving size (43215190) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 34621654 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:31\n",
            "Returning block of 34621655 for bucket 7\n",
            "Getting block 8 of 8\n",
            "  Reserving size (43215190) for bucket 8\n",
            "  Calculating Z arrays for bucket 8\n",
            "  Entering block accumulator loop for bucket 8:\n",
            "  bucket 8: 10%\n",
            "  bucket 8: 20%\n",
            "  bucket 8: 30%\n",
            "  bucket 8: 40%\n",
            "  bucket 8: 50%\n",
            "  bucket 8: 60%\n",
            "  bucket 8: 70%\n",
            "  bucket 8: 80%\n",
            "  bucket 8: 90%\n",
            "  bucket 8: 100%\n",
            "  Sorting block of length 10781597 for bucket 8\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 10781598 for bucket 8\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 67070277\n",
            "fchr[G]: 115125320\n",
            "fchr[T]: 163236848\n",
            "fchr[$]: 230481012\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 81023470 bytes to primary EBWT file: chromosome_index.rev.1.bt2.tmp\n",
            "Wrote 57620260 bytes to secondary EBWT file: chromosome_index.rev.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 230481012\n",
            "    bwtLen: 230481013\n",
            "    sz: 57620253\n",
            "    bwtSz: 57620254\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 14405064\n",
            "    offsSz: 57620256\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 1200422\n",
            "    numLines: 1200422\n",
            "    ebwtTotLen: 76827008\n",
            "    ebwtTotSz: 76827008\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:06:40\n",
            "Renaming chromosome_index.3.bt2.tmp to chromosome_index.3.bt2\n",
            "Renaming chromosome_index.4.bt2.tmp to chromosome_index.4.bt2\n",
            "Renaming chromosome_index.1.bt2.tmp to chromosome_index.1.bt2\n",
            "Renaming chromosome_index.2.bt2.tmp to chromosome_index.2.bt2\n",
            "Renaming chromosome_index.rev.1.bt2.tmp to chromosome_index.rev.1.bt2\n",
            "Renaming chromosome_index.rev.2.bt2.tmp to chromosome_index.rev.2.bt2\n",
            "CPU times: user 3.84 s, sys: 581 ms, total: 4.42 s\n",
            "Wall time: 12min 31s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir bowtie2_res"
      ],
      "metadata": {
        "id": "uDo2ivlXwwuc"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U /content/drive/MyDrive/ENCFF137PVL.fastq \\\n",
        "          -S bowtie2_res/res_ASU.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NclBXSxzwzB0",
        "outputId": "0e21e095-eec6-4754-c00f-ce5990719ef5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "8874861 reads; of these:\n",
            "  8874861 (100.00%) were unpaired; of these:\n",
            "    7262111 (81.83%) aligned 0 times\n",
            "    396647 (4.47%) aligned exactly 1 time\n",
            "    1216103 (13.70%) aligned >1 times\n",
            "18.17% overall alignment rate\n",
            "CPU times: user 2.62 s, sys: 355 ms, total: 2.97 s\n",
            "Wall time: 10min 6s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U /content/drive/MyDrive/ENCFF171DUV.fastq \\\n",
        "          -S bowtie2_res/res_ARK.sam"
      ],
      "metadata": {
        "id": "k_YOk-UL3GqI",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "90ee0f23-dfa3-4e2d-cb5c-669dfd0ad143"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "19825041 reads; of these:\n",
            "  19825041 (100.00%) were unpaired; of these:\n",
            "    16927116 (85.38%) aligned 0 times\n",
            "    588740 (2.97%) aligned exactly 1 time\n",
            "    2309185 (11.65%) aligned >1 times\n",
            "14.62% overall alignment rate\n",
            "CPU times: user 8.34 s, sys: 1.23 s, total: 9.57 s\n",
            "Wall time: 22min 41s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U /content/drive/MyDrive/ENCFF176RJQ.fastq \\\n",
        "          -S bowtie2_res/res_PYU.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "55r_H5ZtH9Rg",
        "outputId": "8d10ca1e-72ec-4907-d340-7b4feda9e4a0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9357767 reads; of these:\n",
            "  9357767 (100.00%) were unpaired; of these:\n",
            "    8283561 (88.52%) aligned 0 times\n",
            "    242561 (2.59%) aligned exactly 1 time\n",
            "    831645 (8.89%) aligned >1 times\n",
            "11.48% overall alignment rate\n",
            "CPU times: user 2.62 s, sys: 374 ms, total: 3 s\n",
            "Wall time: 10min 13s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%cd /content"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "VAj9cIPbT4UU",
        "outputId": "9e2f2217-dad6-42c3-dd67-4fbcda002bf5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir bowtie_res"
      ],
      "metadata": {
        "id": "-cC9kwVET4XD"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%cd /content/bowtie_res"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QN3xXkDdT4bu",
        "outputId": "908007d6-caa9-40cc-90dc-04ca0516f6b0"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/bowtie_res\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -h -S -b -o bowtie_res/res_ASU.bam bowtie_res/res_ASU.sam\n",
        "!samtools view -h -S -b -o bowtie_res/res_ARK.bam bowtie_res/res_ARK.sam"
      ],
      "metadata": {
        "id": "rYGUq45idDMe"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!samtools view -h -S -b -o bowtie_res/res_PYU.bam bowtie2_res/res_PYU.sam"
      ],
      "metadata": {
        "id": "CnqUe_hgeTnQ"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!mv /content/bowtie2_res/res_ASU.sam /content/bowtie_res\n",
        "!mv /content/bowtie2_res/res_ARK.sam /content/bowtie_res"
      ],
      "metadata": {
        "id": "KSJUAmtyUAnK"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "from google.colab import drive\n",
        "drive.mount('/content/drive')\n",
        "\n",
        "!cp bowtie_res/res_ASU.sam /content/drive/MyDrive\n",
        "!cp bowtie_res/res_ARK.sam /content/drive/MyDrive\n",
        "!cp bowtie2_res/res_PYU.sam /content/drive/MyDrive"
      ],
      "metadata": {
        "id": "N2S4VslOTr0j",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "05690a36-b250-43ac-e8ef-3c9d0598d1ec"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount(\"/content/drive\", force_remount=True).\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "KnRxnrTAgms2"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "\n",
        "# 5,6. Filter out blacklisted regions Ð¸ ÑƒÐ´Ð°Ð»ÑÐµÐ¼ PCR Ñ€ÐµÐ¿Ð»Ð¸ÐºÐ¸ Ñ‡ÐµÑ€ÐµÐ· picard removeduplicates\n",
        "ÑÐºÐ°Ñ‡Ð°Ñ‚ÑŒ Ð´Ð»Ñ ÑÐ²Ð¾ÐµÐ³Ð¾ Ð¾Ñ€Ð³Ð°Ð½Ð¸Ð·Ð¼Ð° Ñ ÑÐ°Ð¹Ñ‚Ð° https://github.com/Boyle-Lab/Blacklist\n",
        "\n",
        "Ð¾Ñ‚Ñ„Ð¸Ð»ÑŒÑ‚Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ñ‡Ñ‚ÐµÐ½Ð¸Ñ Ð¸Ð· ÑÑ‚Ð¸Ñ… ÑƒÑ‡Ð°ÑÑ‚ÐºÐ¾Ð²"
      ],
      "metadata": {
        "id": "1h-z-z0tgm7E"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sudo apt update\n",
        "!sudo apt install openjdk-17-jdk"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yi2MPF3KoVQ2",
        "outputId": "cf5ac6e8-e159-48ad-e74e-277312144615"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Get:1 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "Get:2 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "Hit:3 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:6 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Get:7 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [8,730 kB]\n",
            "Hit:8 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease\n",
            "Hit:9 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:10 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [1,369 kB]\n",
            "Hit:11 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:12 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [2,682 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [64.2 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,988 kB]\n",
            "Get:16 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [56.4 kB]\n",
            "Get:17 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [3,774 kB]\n",
            "Get:18 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,668 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,533 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [3,934 kB]\n",
            "Fetched 28.2 MB in 6s (5,104 kB/s)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "32 packages can be upgraded. Run 'apt list --upgradable' to see them.\n",
            "\u001b[1;33mW: \u001b[0mSkipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\u001b[0m\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  fonts-dejavu-core fonts-dejavu-extra libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libfontenc1 libice-dev libsm-dev libxkbfile1\n",
            "  libxt-dev libxtst6 libxxf86dga1 openjdk-17-jdk-headless openjdk-17-jre\n",
            "  openjdk-17-jre-headless x11-utils\n",
            "Suggested packages:\n",
            "  libice-doc libsm-doc libxt-doc openjdk-17-demo openjdk-17-source visualvm\n",
            "  libnss-mdns fonts-ipafont-gothic fonts-ipafont-mincho fonts-wqy-microhei\n",
            "  | fonts-wqy-zenhei fonts-indic mesa-utils\n",
            "The following NEW packages will be installed:\n",
            "  fonts-dejavu-core fonts-dejavu-extra libatk-wrapper-java\n",
            "  libatk-wrapper-java-jni libfontenc1 libice-dev libsm-dev libxkbfile1\n",
            "  libxt-dev libxtst6 libxxf86dga1 openjdk-17-jdk openjdk-17-jdk-headless\n",
            "  openjdk-17-jre openjdk-17-jre-headless x11-utils\n",
            "0 upgraded, 16 newly installed, 0 to remove and 32 not upgraded.\n",
            "Need to get 125 MB of archives.\n",
            "After this operation, 287 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-core all 2.37-2build1 [1,041 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/main amd64 fonts-dejavu-extra all 2.37-2build1 [2,041 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 libfontenc1 amd64 1:1.1.4-1build3 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxkbfile1 amd64 1:1.1.0-1build3 [71.8 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxtst6 amd64 2:1.2.3-1build4 [13.4 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxxf86dga1 amd64 2:1.1.5-0ubuntu3 [12.6 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/main amd64 x11-utils amd64 7.7+5build2 [206 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java all 0.38.0-5build1 [53.1 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu jammy/main amd64 libatk-wrapper-java-jni amd64 0.38.0-5build1 [49.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu jammy/main amd64 libice-dev amd64 2:1.0.10-1build2 [51.4 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu jammy/main amd64 libsm-dev amd64 2:1.2.3-1build2 [18.1 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy/main amd64 libxt-dev amd64 1:1.2.1-1 [396 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre-headless amd64 17.0.14+7-1~22.04.1 [48.3 MB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jre amd64 17.0.14+7-1~22.04.1 [232 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk-headless amd64 17.0.14+7-1~22.04.1 [71.3 MB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 openjdk-17-jdk amd64 17.0.14+7-1~22.04.1 [1,521 kB]\n",
            "Fetched 125 MB in 9s (14.4 MB/s)\n",
            "debconf: unable to initialize frontend: Dialog\n",
            "debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78, <> line 16.)\n",
            "debconf: falling back to frontend: Readline\n",
            "debconf: unable to initialize frontend: Readline\n",
            "debconf: (This frontend requires a controlling tty.)\n",
            "debconf: falling back to frontend: Teletype\n",
            "dpkg-preconfigure: unable to re-open stdin: \n",
            "Selecting previously unselected package fonts-dejavu-core.\n",
            "(Reading database ... 125125 files and directories currently installed.)\n",
            "Preparing to unpack .../00-fonts-dejavu-core_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-core (2.37-2build1) ...\n",
            "Selecting previously unselected package fonts-dejavu-extra.\n",
            "Preparing to unpack .../01-fonts-dejavu-extra_2.37-2build1_all.deb ...\n",
            "Unpacking fonts-dejavu-extra (2.37-2build1) ...\n",
            "Selecting previously unselected package libfontenc1:amd64.\n",
            "Preparing to unpack .../02-libfontenc1_1%3a1.1.4-1build3_amd64.deb ...\n",
            "Unpacking libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Selecting previously unselected package libxkbfile1:amd64.\n",
            "Preparing to unpack .../03-libxkbfile1_1%3a1.1.0-1build3_amd64.deb ...\n",
            "Unpacking libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Selecting previously unselected package libxtst6:amd64.\n",
            "Preparing to unpack .../04-libxtst6_2%3a1.2.3-1build4_amd64.deb ...\n",
            "Unpacking libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Selecting previously unselected package libxxf86dga1:amd64.\n",
            "Preparing to unpack .../05-libxxf86dga1_2%3a1.1.5-0ubuntu3_amd64.deb ...\n",
            "Unpacking libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Selecting previously unselected package x11-utils.\n",
            "Preparing to unpack .../06-x11-utils_7.7+5build2_amd64.deb ...\n",
            "Unpacking x11-utils (7.7+5build2) ...\n",
            "Selecting previously unselected package libatk-wrapper-java.\n",
            "Preparing to unpack .../07-libatk-wrapper-java_0.38.0-5build1_all.deb ...\n",
            "Unpacking libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libatk-wrapper-java-jni:amd64.\n",
            "Preparing to unpack .../08-libatk-wrapper-java-jni_0.38.0-5build1_amd64.deb ...\n",
            "Unpacking libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Selecting previously unselected package libice-dev:amd64.\n",
            "Preparing to unpack .../09-libice-dev_2%3a1.0.10-1build2_amd64.deb ...\n",
            "Unpacking libice-dev:amd64 (2:1.0.10-1build2) ...\n",
            "Selecting previously unselected package libsm-dev:amd64.\n",
            "Preparing to unpack .../10-libsm-dev_2%3a1.2.3-1build2_amd64.deb ...\n",
            "Unpacking libsm-dev:amd64 (2:1.2.3-1build2) ...\n",
            "Selecting previously unselected package libxt-dev:amd64.\n",
            "Preparing to unpack .../11-libxt-dev_1%3a1.2.1-1_amd64.deb ...\n",
            "Unpacking libxt-dev:amd64 (1:1.2.1-1) ...\n",
            "Selecting previously unselected package openjdk-17-jre-headless:amd64.\n",
            "Preparing to unpack .../12-openjdk-17-jre-headless_17.0.14+7-1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jre-headless:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "Selecting previously unselected package openjdk-17-jre:amd64.\n",
            "Preparing to unpack .../13-openjdk-17-jre_17.0.14+7-1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jre:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "Selecting previously unselected package openjdk-17-jdk-headless:amd64.\n",
            "Preparing to unpack .../14-openjdk-17-jdk-headless_17.0.14+7-1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jdk-headless:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "Selecting previously unselected package openjdk-17-jdk:amd64.\n",
            "Preparing to unpack .../15-openjdk-17-jdk_17.0.14+7-1~22.04.1_amd64.deb ...\n",
            "Unpacking openjdk-17-jdk:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "Setting up libice-dev:amd64 (2:1.0.10-1build2) ...\n",
            "Setting up libsm-dev:amd64 (2:1.2.3-1build2) ...\n",
            "Setting up libxtst6:amd64 (2:1.2.3-1build4) ...\n",
            "Setting up libxxf86dga1:amd64 (2:1.1.5-0ubuntu3) ...\n",
            "Setting up libfontenc1:amd64 (1:1.1.4-1build3) ...\n",
            "Setting up libxt-dev:amd64 (1:1.2.1-1) ...\n",
            "Setting up fonts-dejavu-core (2.37-2build1) ...\n",
            "Setting up openjdk-17-jre-headless:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/java to provide /usr/bin/java (java) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jpackage to provide /usr/bin/jpackage (jpackage) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/keytool to provide /usr/bin/keytool (keytool) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/rmiregistry to provide /usr/bin/rmiregistry (rmiregistry) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/lib/jexec to provide /usr/bin/jexec (jexec) in auto mode\n",
            "Setting up fonts-dejavu-extra (2.37-2build1) ...\n",
            "Setting up libxkbfile1:amd64 (1:1.1.0-1build3) ...\n",
            "Setting up openjdk-17-jre:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "Setting up openjdk-17-jdk-headless:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jar to provide /usr/bin/jar (jar) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jarsigner to provide /usr/bin/jarsigner (jarsigner) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javac to provide /usr/bin/javac (javac) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javadoc to provide /usr/bin/javadoc (javadoc) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/javap to provide /usr/bin/javap (javap) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jcmd to provide /usr/bin/jcmd (jcmd) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdb to provide /usr/bin/jdb (jdb) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeprscan to provide /usr/bin/jdeprscan (jdeprscan) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jdeps to provide /usr/bin/jdeps (jdeps) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jfr to provide /usr/bin/jfr (jfr) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jimage to provide /usr/bin/jimage (jimage) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jinfo to provide /usr/bin/jinfo (jinfo) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jlink to provide /usr/bin/jlink (jlink) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmap to provide /usr/bin/jmap (jmap) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jmod to provide /usr/bin/jmod (jmod) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jps to provide /usr/bin/jps (jps) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jrunscript to provide /usr/bin/jrunscript (jrunscript) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jshell to provide /usr/bin/jshell (jshell) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstack to provide /usr/bin/jstack (jstack) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstat to provide /usr/bin/jstat (jstat) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jstatd to provide /usr/bin/jstatd (jstatd) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/serialver to provide /usr/bin/serialver (serialver) in auto mode\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jhsdb to provide /usr/bin/jhsdb (jhsdb) in auto mode\n",
            "Setting up x11-utils (7.7+5build2) ...\n",
            "Setting up libatk-wrapper-java (0.38.0-5build1) ...\n",
            "Setting up openjdk-17-jdk:amd64 (17.0.14+7-1~22.04.1) ...\n",
            "update-alternatives: using /usr/lib/jvm/java-17-openjdk-amd64/bin/jconsole to provide /usr/bin/jconsole (jconsole) in auto mode\n",
            "Setting up libatk-wrapper-java-jni:amd64 (0.38.0-5build1) ...\n",
            "Processing triggers for fontconfig (2.13.1-4.2ubuntu5) ...\n",
            "Processing triggers for hicolor-icon-theme (0.17-2) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/liblzma.so.5 is not a symbolic link\n",
            "\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for mailcap (3.70+nmu1ubuntu1) ...\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://raw.githubusercontent.com/Boyle-Lab/Blacklist/refs/heads/master/lists/hg38-blacklist.v2.bed.gz\n",
        "!gunzip hg38-blacklist.v2.bed.gz\n",
        "\n",
        "!bedtools intersect -a bowtie_res/res_ASU.bam -b hg38-blacklist.v2.bed -v > bowtie_res/res_ASU_bl.bam\n",
        "\n",
        "!samtools sort bowtie_res/res_ASU_bl.bam -o bowtie_res/res_ASU_bl_sorted.bam\n",
        "\n",
        "!wget https://github.com/broadinstitute/picard/releases/download/3.3.0/picard.jar\n",
        "\n",
        "!/usr/lib/jvm/java-17-openjdk-amd64/bin/java -jar picard.jar MarkDuplicates REMOVE_DUPLICATES=true I=bowtie_res/res_ASU_bl_sorted.bam O=bowtie_res/res_ASU_bl_sorted_rd.bam M=bowtie_res/ASU_metrices.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1NwY2SF1grLj",
        "outputId": "cfc66abe-f228-4996-c8ac-7bd7d64458e6"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-03-08 16:46:43--  https://raw.githubusercontent.com/Boyle-Lab/Blacklist/refs/heads/master/lists/hg38-blacklist.v2.bed.gz\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.108.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 5867 (5.7K) [application/octet-stream]\n",
            "Saving to: â€˜hg38-blacklist.v2.bed.gzâ€™\n",
            "\n",
            "hg38-blacklist.v2.b 100%[===================>]   5.73K  --.-KB/s    in 0s      \n",
            "\n",
            "2025-03-08 16:46:43 (39.1 MB/s) - â€˜hg38-blacklist.v2.bed.gzâ€™ saved [5867/5867]\n",
            "\n",
            "gzip: hg38-blacklist.v2.bed already exists; do you wish to overwrite (y or n)? y\n",
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
            "--2025-03-08 16:49:15--  https://github.com/broadinstitute/picard/releases/download/3.3.0/picard.jar\n",
            "Resolving github.com (github.com)... 20.27.177.113\n",
            "Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/6c90b3fd-a077-4677-9fc1-4078562f6938?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250308T164915Z&X-Amz-Expires=300&X-Amz-Signature=7883b61c52db77f7d9afc7a4ca76c118c8e1b2e040523ab8f3b2dc675589ebe5&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-03-08 16:49:15--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/6c90b3fd-a077-4677-9fc1-4078562f6938?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250308T164915Z&X-Amz-Expires=300&X-Amz-Signature=7883b61c52db77f7d9afc7a4ca76c118c8e1b2e040523ab8f3b2dc675589ebe5&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.109.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 63050907 (60M) [application/octet-stream]\n",
            "Saving to: â€˜picard.jar.5â€™\n",
            "\n",
            "picard.jar.5        100%[===================>]  60.13M  63.1MB/s    in 1.0s    \n",
            "\n",
            "2025-03-08 16:49:18 (63.1 MB/s) - â€˜picard.jar.5â€™ saved [63050907/63050907]\n",
            "\n",
            "INFO\t2025-03-08 16:49:20\tMarkDuplicates\t\n",
            "\n",
            "********** NOTE: Picard's command line syntax is changing.\n",
            "**********\n",
            "********** For more information, please see:\n",
            "********** \n",
            "https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
            "**********\n",
            "********** The command line looks like this in the new syntax:\n",
            "**********\n",
            "**********    MarkDuplicates -REMOVE_DUPLICATES true -I bowtie_res/res_ASU_bl_sorted.bam -O bowtie_res/res_ASU_bl_sorted_rd.bam -M bowtie_res/ASU_metrices.txt\n",
            "**********\n",
            "\n",
            "\n",
            "16:49:20.983 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Sat Mar 08 16:49:21 UTC 2025] MarkDuplicates INPUT=[bowtie_res/res_ASU_bl_sorted.bam] OUTPUT=bowtie_res/res_ASU_bl_sorted_rd.bam METRICS_FILE=bowtie_res/ASU_metrices.txt REMOVE_DUPLICATES=true    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 TAG_DUPLICATE_SET_MEMBERS=false REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag CLEAR_DT=true DUPLEX_UMI=false FLOW_MODE=false FLOW_DUP_STRATEGY=FLOW_QUALITY_SUM_STRATEGY FLOW_USE_END_IN_UNPAIRED_READS=false FLOW_USE_UNPAIRED_CLIPPED_END=false FLOW_UNPAIRED_END_UNCERTAINTY=0 FLOW_UNPAIRED_START_UNCERTAINTY=0 FLOW_SKIP_FIRST_N_FLOWS=0 FLOW_Q_IS_KNOWN_END=false FLOW_EFFECTIVE_QUALITY_THRESHOLD=15 ADD_PG_TAG_TO_READS=true ASSUME_SORTED=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last three ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 MAX_OPTICAL_DUPLICATE_SET_SIZE=300000 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
            "[Sat Mar 08 16:49:21 UTC 2025] Executing as root@07594b4d306b on Linux 6.1.85+ amd64; OpenJDK 64-Bit Server VM 17.0.14+7-Ubuntu-122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: 3.3.0\n",
            "INFO\t2025-03-08 16:49:21\tMarkDuplicates\tStart of doWork freeMemory: 38135056; totalMemory: 50331648; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:49:21\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-03-08 16:49:21\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "INFO\t2025-03-08 16:49:26\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    4s.  Last read position: chr14:76,884,365\n",
            "INFO\t2025-03-08 16:49:26\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2025-03-08 16:49:28\tMarkDuplicates\tRead 1439613 records. 0 pairs never matched.\n",
            "INFO\t2025-03-08 16:49:28\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 516630992; totalMemory: 734003200; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:49:28\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 719208072; totalMemory: 1583349760; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tMarking 184753 records as duplicates.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "INFO\t2025-03-08 16:49:29\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tBefore output close freeMemory: 37826032; totalMemory: 50331648; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-03-08 16:50:40\tMarkDuplicates\tAfter output close freeMemory: 37826032; totalMemory: 50331648; maxMemory: 3403677696\n",
            "[Sat Mar 08 16:50:40 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 1.33 minutes.\n",
            "Runtime.totalMemory()=50331648\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools intersect -a bowtie_res/res_ARK.bam -b hg38-blacklist.v2.bed -v > bowtie_res/res_ARK_bl.bam\n",
        "\n",
        "!samtools sort bowtie_res/res_ARK_bl.bam -o bowtie_res/res_ARK_bl_sorted.bam\n",
        "\n",
        "!wget https://github.com/broadinstitute/picard/releases/download/3.3.0/picard.jar\n",
        "\n",
        "!/usr/lib/jvm/java-17-openjdk-amd64/bin/java -jar picard.jar MarkDuplicates REMOVE_DUPLICATES=true I=bowtie_res/res_ARK_bl_sorted.bam O=bowtie_res/res_ARK_bl_sorted_rd.bam M=bowtie_res/ARK_metrices.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1JT6z7iDgrPv",
        "outputId": "09ee8f88-350f-4788-d068-d924648f25dd"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 4 files and 1 in-memory blocks...\n",
            "--2025-03-08 16:57:56--  https://github.com/broadinstitute/picard/releases/download/3.3.0/picard.jar\n",
            "Resolving github.com (github.com)... 20.27.177.113\n",
            "Connecting to github.com (github.com)|20.27.177.113|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/6c90b3fd-a077-4677-9fc1-4078562f6938?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250308T165756Z&X-Amz-Expires=300&X-Amz-Signature=7fbd5a4349d277ed503566387f765d60dcc88707e665b1540a9db87d00f2360e&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream [following]\n",
            "--2025-03-08 16:57:57--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/18225913/6c90b3fd-a077-4677-9fc1-4078562f6938?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=releaseassetproduction%2F20250308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20250308T165756Z&X-Amz-Expires=300&X-Amz-Signature=7fbd5a4349d277ed503566387f765d60dcc88707e665b1540a9db87d00f2360e&X-Amz-SignedHeaders=host&response-content-disposition=attachment%3B%20filename%3Dpicard.jar&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.110.133, 185.199.111.133, 185.199.109.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.110.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 63050907 (60M) [application/octet-stream]\n",
            "Saving to: â€˜picard.jar.6â€™\n",
            "\n",
            "picard.jar.6        100%[===================>]  60.13M  65.9MB/s    in 0.9s    \n",
            "\n",
            "2025-03-08 16:57:59 (65.9 MB/s) - â€˜picard.jar.6â€™ saved [63050907/63050907]\n",
            "\n",
            "INFO\t2025-03-08 16:58:01\tMarkDuplicates\t\n",
            "\n",
            "********** NOTE: Picard's command line syntax is changing.\n",
            "**********\n",
            "********** For more information, please see:\n",
            "********** \n",
            "https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
            "**********\n",
            "********** The command line looks like this in the new syntax:\n",
            "**********\n",
            "**********    MarkDuplicates -REMOVE_DUPLICATES true -I bowtie_res/res_ARK_bl_sorted.bam -O bowtie_res/res_ARK_bl_sorted_rd.bam -M bowtie_res/ARK_metrices.txt\n",
            "**********\n",
            "\n",
            "\n",
            "16:58:02.039 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Sat Mar 08 16:58:02 UTC 2025] MarkDuplicates INPUT=[bowtie_res/res_ARK_bl_sorted.bam] OUTPUT=bowtie_res/res_ARK_bl_sorted_rd.bam METRICS_FILE=bowtie_res/ARK_metrices.txt REMOVE_DUPLICATES=true    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 TAG_DUPLICATE_SET_MEMBERS=false REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag CLEAR_DT=true DUPLEX_UMI=false FLOW_MODE=false FLOW_DUP_STRATEGY=FLOW_QUALITY_SUM_STRATEGY FLOW_USE_END_IN_UNPAIRED_READS=false FLOW_USE_UNPAIRED_CLIPPED_END=false FLOW_UNPAIRED_END_UNCERTAINTY=0 FLOW_UNPAIRED_START_UNCERTAINTY=0 FLOW_SKIP_FIRST_N_FLOWS=0 FLOW_Q_IS_KNOWN_END=false FLOW_EFFECTIVE_QUALITY_THRESHOLD=15 ADD_PG_TAG_TO_READS=true ASSUME_SORTED=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last three ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 MAX_OPTICAL_DUPLICATE_SET_SIZE=300000 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
            "[Sat Mar 08 16:58:02 UTC 2025] Executing as root@07594b4d306b on Linux 6.1.85+ amd64; OpenJDK 64-Bit Server VM 17.0.14+7-Ubuntu-122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: 3.3.0\n",
            "INFO\t2025-03-08 16:58:02\tMarkDuplicates\tStart of doWork freeMemory: 38129800; totalMemory: 50331648; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:58:02\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-03-08 16:58:02\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "INFO\t2025-03-08 16:58:07\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    4s.  Last read position: chr14:54,360,178\n",
            "INFO\t2025-03-08 16:58:07\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2025-03-08 16:58:09\tMarkDuplicates\tRead     2,000,000 records.  Elapsed time: 00:00:06s.  Time for last 1,000,000:    2s.  Last read position: chr14:83,958,133\n",
            "INFO\t2025-03-08 16:58:09\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2025-03-08 16:58:11\tMarkDuplicates\tRead 2731907 records. 0 pairs never matched.\n",
            "INFO\t2025-03-08 16:58:11\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 738292608; totalMemory: 1048576000; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:58:11\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-03-08 16:58:12\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-03-08 16:58:12\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-03-08 16:58:13\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-03-08 16:58:13\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 1033771096; totalMemory: 1897922560; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 16:58:13\tMarkDuplicates\tMarking 329941 records as duplicates.\n",
            "INFO\t2025-03-08 16:58:13\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "INFO\t2025-03-08 16:58:13\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-03-08 16:59:49\tMarkDuplicates\tWritten    10,000,000 records.  Elapsed time: 00:01:36s.  Time for last 10,000,000:   96s.  Last read position: */*\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tBefore output close freeMemory: 43980456; totalMemory: 56623104; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-03-08 17:00:58\tMarkDuplicates\tAfter output close freeMemory: 43980456; totalMemory: 56623104; maxMemory: 3403677696\n",
            "[Sat Mar 08 17:00:58 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 2.94 minutes.\n",
            "Runtime.totalMemory()=56623104\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/broadinstitute/picard/releases/download/3.3.0/picard.jar\n",
        "\n",
        "!sudo apt update\n",
        "!sudo apt install openjdk-17-jdk\n",
        "\n",
        "!/usr/lib/jvm/java-17-openjdk-amd64/bin/java -jar picard.jar MarkDuplicates \\\n",
        "    REMOVE_DUPLICATES=true I=intersect/res_ERN_sorted.bam \\\n",
        "    O=intersect/res_ERN_filtered.bam M=intersect/ERN_metrices.txt"
      ],
      "metadata": {
        "id": "NDegDc2q1xLB"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bedtools intersect -a bowtie_res/res_PYU.bam -b hg38-blacklist.v2.bed -v > bowtie_res/res_PYU_bl.bam\n",
        "\n",
        "!samtools sort bowtie_res/res_PYU_bl.bam -o bowtie_res/res_PYU_bl_sorted.bam\n",
        "\n",
        "!/usr/lib/jvm/java-17-openjdk-amd64/bin/java -jar picard.jar MarkDuplicates REMOVE_DUPLICATES=true I=bowtie_res/res_PYU_bl_sorted.bam O=bowtie_res/res_PYU_bl_sorted_rd.bam M=bowtie_res/PYU_metrices.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0Dmt869_lriy",
        "outputId": "b12b17c8-21d3-4a38-ed39-b1e0d2976ef5"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[bam_sort_core] merging from 1 files and 1 in-memory blocks...\n",
            "INFO\t2025-03-08 17:04:46\tMarkDuplicates\t\n",
            "\n",
            "********** NOTE: Picard's command line syntax is changing.\n",
            "**********\n",
            "********** For more information, please see:\n",
            "********** \n",
            "https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)\n",
            "**********\n",
            "********** The command line looks like this in the new syntax:\n",
            "**********\n",
            "**********    MarkDuplicates -REMOVE_DUPLICATES true -I bowtie_res/res_PYU_bl_sorted.bam -O bowtie_res/res_PYU_bl_sorted_rd.bam -M bowtie_res/PYU_metrices.txt\n",
            "**********\n",
            "\n",
            "\n",
            "17:04:46.855 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/content/picard.jar!/com/intel/gkl/native/libgkl_compression.so\n",
            "[Sat Mar 08 17:04:46 UTC 2025] MarkDuplicates INPUT=[bowtie_res/res_PYU_bl_sorted.bam] OUTPUT=bowtie_res/res_PYU_bl_sorted_rd.bam METRICS_FILE=bowtie_res/PYU_metrices.txt REMOVE_DUPLICATES=true    MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=50000 MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=8000 SORTING_COLLECTION_SIZE_RATIO=0.25 TAG_DUPLICATE_SET_MEMBERS=false REMOVE_SEQUENCING_DUPLICATES=false TAGGING_POLICY=DontTag CLEAR_DT=true DUPLEX_UMI=false FLOW_MODE=false FLOW_DUP_STRATEGY=FLOW_QUALITY_SUM_STRATEGY FLOW_USE_END_IN_UNPAIRED_READS=false FLOW_USE_UNPAIRED_CLIPPED_END=false FLOW_UNPAIRED_END_UNCERTAINTY=0 FLOW_UNPAIRED_START_UNCERTAINTY=0 FLOW_SKIP_FIRST_N_FLOWS=0 FLOW_Q_IS_KNOWN_END=false FLOW_EFFECTIVE_QUALITY_THRESHOLD=15 ADD_PG_TAG_TO_READS=true ASSUME_SORTED=false DUPLICATE_SCORING_STRATEGY=SUM_OF_BASE_QUALITIES PROGRAM_RECORD_ID=MarkDuplicates PROGRAM_GROUP_NAME=MarkDuplicates READ_NAME_REGEX=<optimized capture of last three ':' separated fields as numeric values> OPTICAL_DUPLICATE_PIXEL_DISTANCE=100 MAX_OPTICAL_DUPLICATE_SET_SIZE=300000 VERBOSITY=INFO QUIET=false VALIDATION_STRINGENCY=STRICT COMPRESSION_LEVEL=5 MAX_RECORDS_IN_RAM=500000 CREATE_INDEX=false CREATE_MD5_FILE=false USE_JDK_DEFLATER=false USE_JDK_INFLATER=false\n",
            "[Sat Mar 08 17:04:47 UTC 2025] Executing as root@07594b4d306b on Linux 6.1.85+ amd64; OpenJDK 64-Bit Server VM 17.0.14+7-Ubuntu-122.04.1; Deflater: Intel; Inflater: Intel; Provider GCS is available; Picard version: 3.3.0\n",
            "INFO\t2025-03-08 17:04:47\tMarkDuplicates\tStart of doWork freeMemory: 44273296; totalMemory: 56623104; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 17:04:47\tMarkDuplicates\tReading input file and constructing read end information.\n",
            "INFO\t2025-03-08 17:04:47\tMarkDuplicates\tWill retain up to 12332165 data points before spilling to disk.\n",
            "INFO\t2025-03-08 17:04:51\tMarkDuplicates\tRead     1,000,000 records.  Elapsed time: 00:00:04s.  Time for last 1,000,000:    4s.  Last read position: chr14:105,311,176\n",
            "INFO\t2025-03-08 17:04:51\tMarkDuplicates\tTracking 0 as yet unmatched pairs. 0 records in RAM.\n",
            "INFO\t2025-03-08 17:04:51\tMarkDuplicates\tRead 1011306 records. 0 pairs never matched.\n",
            "INFO\t2025-03-08 17:04:51\tMarkDuplicates\tAfter buildSortedReadEndLists freeMemory: 442445672; totalMemory: 629145600; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 17:04:51\tMarkDuplicates\tWill retain up to 106364928 duplicate indices before spilling to disk.\n",
            "INFO\t2025-03-08 17:04:52\tMarkDuplicates\tTraversing read pair information and detecting duplicates.\n",
            "INFO\t2025-03-08 17:04:52\tMarkDuplicates\tTraversing fragment information and detecting duplicates.\n",
            "INFO\t2025-03-08 17:04:53\tMarkDuplicates\tSorting list of duplicate records.\n",
            "INFO\t2025-03-08 17:04:53\tMarkDuplicates\tAfter generateDuplicateIndexes freeMemory: 614361128; totalMemory: 1478492160; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 17:04:53\tMarkDuplicates\tMarking 74819 records as duplicates.\n",
            "INFO\t2025-03-08 17:04:53\tMarkDuplicates\tFound 0 optical duplicate clusters.\n",
            "INFO\t2025-03-08 17:04:53\tMarkDuplicates\tReads are assumed to be ordered by: coordinate\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tWriting complete. Closing input iterator.\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tDuplicate Index cleanup.\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tGetting Memory Stats.\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tBefore output close freeMemory: 44001872; totalMemory: 56623104; maxMemory: 3403677696\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tClosed outputs. Getting more Memory Stats.\n",
            "INFO\t2025-03-08 17:06:04\tMarkDuplicates\tAfter output close freeMemory: 37794152; totalMemory: 50331648; maxMemory: 3403677696\n",
            "[Sat Mar 08 17:06:04 UTC 2025] picard.sam.markduplicates.MarkDuplicates done. Elapsed time: 1.30 minutes.\n",
            "Runtime.totalMemory()=50331648\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "##7.  Peak calling\n",
        "\n",
        "ÐÐ° Ð¾ÑÐ½Ð¾Ð²Ðµ Ð¾Ñ‚Ñ„Ð¸Ð»ÑŒÑ‚Ñ€Ð¾Ð²Ð°Ð½Ñ‹Ñ… Ñ„Ð°Ð¹Ð»Ð¾Ð² Ð¿Ð¾Ð»ÑƒÑ‹Ñ‡Ð°ÐµÐ¼ Ð¿Ð¸ÐºÐ¸ \\\n",
        "ÐŸÐ¾Ð¼Ð¸Ð¼Ð¾ macs2 Ð¼Ð¾Ð¶ÐµÑ‚ Ð±Ñ‹Ñ‚ÑŒ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½ Ð¿Ð°ÐºÐµÑ‚ Homer Ð´Ð»Ñ Ð¿Ð¾Ð¸ÑÐºÐ° Ð¿Ð¸ÐºÐ¾Ð², Ð° Ñ‚Ð°ÐºÐ¶Ðµ macs3."
      ],
      "metadata": {
        "id": "5GYpUczSsupK"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir macs2"
      ],
      "metadata": {
        "id": "2CB5Ik2TRJ7M"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t /content/bowtie_res/res_ASU_bl_sorted_rd.bam \\\n",
        "    -c /content/bowtie_res/res_PYU_bl_sorted_rd.bam \\\n",
        " \t  -f BAM \\\n",
        "\t  --outdir macs2 \\\n",
        "    -n ASU"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "0mOfy-cbtyog",
        "outputId": "814a0762-b747-42d4-908f-70ba55686fac"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Sat, 08 Mar 2025 18:02:05: \n",
            "# Command line: callpeak --broad -t /content/bowtie_res/res_ASU_bl_sorted_rd.bam -c /content/bowtie_res/res_PYU_bl_sorted_rd.bam -f BAM --outdir macs2 -n ASU\n",
            "# ARGUMENTS LIST:\n",
            "# name = ASU\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['/content/bowtie_res/res_ASU_bl_sorted_rd.bam']\n",
            "# control file = ['/content/bowtie_res/res_PYU_bl_sorted_rd.bam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:05: #1 read tag files... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:05: #1 read treatment tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:09:  1000000 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:21: 1254860 reads have been read. \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:21: #1.2 read input tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: 936487 reads have been read. \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 tag size is determined as 36 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 tag size = 36.0 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  total tags in treatment: 1254860 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 user defined the maximum tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  tags after filtering in treatment: 1254860 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  Redundant rate of treatment: 0.00 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  total tags in control: 936487 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 user defined the maximum tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  tags after filtering in control: 936487 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1  Redundant rate of control: 0.00 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #1 finished! \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 Build Peak Model... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 number of paired peaks: 13410 \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: start model_add_line... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: start X-correlation... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: end of X-cor \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 finished! \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 predicted fragment length is 36 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2 alternative fragment length(s) may be 36,106,478 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #2.2 Generate R script for model : macs2/ASU_model.r \n",
            "WARNING @ Sat, 08 Mar 2025 18:02:37: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Sat, 08 Mar 2025 18:02:37: #2 You may need to consider one of the other alternative d(s): 36,106,478 \n",
            "WARNING @ Sat, 08 Mar 2025 18:02:37: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #3 Call peaks... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:37: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:47: #3 Call peaks for each chromosome... \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:49: #4 Write output xls file... macs2/ASU_peaks.xls \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:49: #4 Write broad peak in broadPeak format file... macs2/ASU_peaks.broadPeak \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:49: #4 Write broad peak in bed12/gappedPeak format file... macs2/ASU_peaks.gappedPeak \n",
            "INFO  @ Sat, 08 Mar 2025 18:02:49: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t /content/bowtie_res/res_ARK_bl_sorted_rd.bam \\\n",
        "    -c /content/bowtie_res/res_PYU_bl_sorted_rd.bam \\\n",
        " \t  -f BAM \\\n",
        "\t  --outdir macs2 \\\n",
        "    -n ARK"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AEg3b5N3tyrA",
        "outputId": "8a958fb4-1d26-498b-c832-5762c8063ebb"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Sat, 08 Mar 2025 18:06:19: \n",
            "# Command line: callpeak --broad -t /content/bowtie_res/res_ARK_bl_sorted_rd.bam -c /content/bowtie_res/res_PYU_bl_sorted_rd.bam -f BAM --outdir macs2 -n ARK\n",
            "# ARGUMENTS LIST:\n",
            "# name = ARK\n",
            "# format = BAM\n",
            "# ChIP-seq file = ['/content/bowtie_res/res_ARK_bl_sorted_rd.bam']\n",
            "# control file = ['/content/bowtie_res/res_PYU_bl_sorted_rd.bam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:19: #1 read tag files... \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:19: #1 read treatment tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:21:  1000000 \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:23:  2000000 \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:54: 2401966 reads have been read. \n",
            "INFO  @ Sat, 08 Mar 2025 18:06:54: #1.2 read input tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: 936487 reads have been read. \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 tag size is determined as 36 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 tag size = 36.0 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  total tags in treatment: 2401966 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 user defined the maximum tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  tags after filtering in treatment: 2401966 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  Redundant rate of treatment: 0.00 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  total tags in control: 936487 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 user defined the maximum tags... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  tags after filtering in control: 936487 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1  Redundant rate of control: 0.00 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #1 finished! \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #2 Build Peak Model... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:09: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:10: #2 number of paired peaks: 16898 \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:10: start model_add_line... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: start X-correlation... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: end of X-cor \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #2 finished! \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #2 predicted fragment length is 39 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #2 alternative fragment length(s) may be 39 bps \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #2.2 Generate R script for model : macs2/ARK_model.r \n",
            "WARNING @ Sat, 08 Mar 2025 18:07:11: #2 Since the d (39) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Sat, 08 Mar 2025 18:07:11: #2 You may need to consider one of the other alternative d(s): 39 \n",
            "WARNING @ Sat, 08 Mar 2025 18:07:11: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #3 Call peaks... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:11: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:19: #3 Call peaks for each chromosome... \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:22: #4 Write output xls file... macs2/ARK_peaks.xls \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:22: #4 Write broad peak in broadPeak format file... macs2/ARK_peaks.broadPeak \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:22: #4 Write broad peak in bed12/gappedPeak format file... macs2/ARK_peaks.gappedPeak \n",
            "INFO  @ Sat, 08 Mar 2025 18:07:22: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF367KIF/@@download/ENCFF367KIF.bed.gz\n",
        "!gzip -d ENCFF367KIF.bed.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "4A8VBpTA8qKc",
        "outputId": "33882732-40af-432f-f0aa-5557e9b6ec5f"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-03-08 18:21:02--  https://www.encodeproject.org/files/ENCFF367KIF/@@download/ENCFF367KIF.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2017/01/05/b7059514-141c-4366-ad03-43b74eb22297/ENCFF367KIF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF367KIF.bed.gz&AWSAccessKeyId=ASIATGZNGCNX26SYBQIN&Signature=8rP2f1qhkjKGb1kzwA9i%2FqR0HaU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEBoaCXVzLXdlc3QtMiJIMEYCIQDToRAXfyEIv0xR0yy7dY0gcLCepCtmzalfdj2o7BGJOwIhAPUPbiyFws9l99X1CWFbFqqPF%2FdVEGUl%2F0ACYbvvyKduKrMFCGMQABoMMjIwNzQ4NzE0ODYzIgwFcy0M%2BSJ0ykfcUCgqkAWtWyFZ%2FYnh3nPeNvu5FNY7q65mSpoh7X6f20kY4gKYkPWv1N3uo85NFbW4eWgi0UXdPlJPacP0ZJJAJywftSkoQHL7dbq36f4r%2B1%2BHIa2D3HBIHLUrKEAC18gjaHnVKxA2szY%2F%2BU%2B9TVBc%2FmMXYWQ85GfoKLusD6xqEpdXdZyYEVjOxyJSZRPrwNrGT2fNqU7E%2Bzm4Zszo%2Ft37IzASATY8DnQWLnJ1vOEidSx1843b39MrfVJXcc2I0q9zQq25fLV%2FGp6gmiUDQx5viS%2Bz8P9mokOC708%2B94qSrWXWD9IZvT6ZV3ZV31nnsJWVIKFZPc4n7jqD9AT6C%2BFAjIb3GVPCIm6IDSxVugS3EhmbW%2BG4V4JyVP7bfuwf9AlkM%2Fw9yFzr9S8qecJY233%2FPi0z2qpKNZrW%2B5O904D3y7JShxWwED5LgSU1APwHApRBqlbYoWGoptz6%2BfIA0Ds8oUdd1BYKyf1Y8IIUsbap84AhK9WoEbbFtNVAAiPpbg3FTGPqDXvS8jIXnbyJuvpTIulppMHE56N4m%2FnzFmY%2FZmO6kBx47qXAVkHtYPjDzXZ8p0B%2BrgI9WZq%2FfZOGAOq4VcuXxmEuS0yu52Bn8m2ImSyE02bPeSLDBzEcBRfGEdhE%2Ff1293GGIvUhUvSqU%2B6E6Q3oGsuuU1v9l%2F9cXc95fZdd%2FBXuzYqCSN7ucHGvlnbTNNbLjrzzW3cBY1bLiZmFanxHTitffCbZupEbtYJQC8SMKyYsBop0T6k4NcsMRGByAgZKqdbhLOrKMezveQO3xsurXwTzbvVnN%2FKUGySWcFhP5fwkhuhFYMItiYO8MfXfKa57Pdai3%2BM0DoNEb80HfZEhL5%2B8qdSAFCApFvtXwcWswb4IsDDJkrK%2BBjqwAR50inE6zABnUAMaleyf2wNh4E2Ad6PKC8xMXekh3yB8E8jnsSYfnsIJXOfo6duyAupaIjmHigDtYafRb47S%2B6Y4kzbtMjcahW7AV%2BGQ13dczT8py2n705y2TYD8U8e0PU9%2Bj9tHqEil9f5vgzpL3aY9LvDhKWGcLpQwOZQE%2FFU%2B%2FNchhzCKM%2BH0s%2F5t66fbuBgvgW9o5WNW4raRHLgB1SoxmE3OnTm0mQdnnfAEKEX%2B&Expires=1741587663 [following]\n",
            "--2025-03-08 18:21:03--  https://encode-public.s3.amazonaws.com/2017/01/05/b7059514-141c-4366-ad03-43b74eb22297/ENCFF367KIF.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF367KIF.bed.gz&AWSAccessKeyId=ASIATGZNGCNX26SYBQIN&Signature=8rP2f1qhkjKGb1kzwA9i%2FqR0HaU%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEBoaCXVzLXdlc3QtMiJIMEYCIQDToRAXfyEIv0xR0yy7dY0gcLCepCtmzalfdj2o7BGJOwIhAPUPbiyFws9l99X1CWFbFqqPF%2FdVEGUl%2F0ACYbvvyKduKrMFCGMQABoMMjIwNzQ4NzE0ODYzIgwFcy0M%2BSJ0ykfcUCgqkAWtWyFZ%2FYnh3nPeNvu5FNY7q65mSpoh7X6f20kY4gKYkPWv1N3uo85NFbW4eWgi0UXdPlJPacP0ZJJAJywftSkoQHL7dbq36f4r%2B1%2BHIa2D3HBIHLUrKEAC18gjaHnVKxA2szY%2F%2BU%2B9TVBc%2FmMXYWQ85GfoKLusD6xqEpdXdZyYEVjOxyJSZRPrwNrGT2fNqU7E%2Bzm4Zszo%2Ft37IzASATY8DnQWLnJ1vOEidSx1843b39MrfVJXcc2I0q9zQq25fLV%2FGp6gmiUDQx5viS%2Bz8P9mokOC708%2B94qSrWXWD9IZvT6ZV3ZV31nnsJWVIKFZPc4n7jqD9AT6C%2BFAjIb3GVPCIm6IDSxVugS3EhmbW%2BG4V4JyVP7bfuwf9AlkM%2Fw9yFzr9S8qecJY233%2FPi0z2qpKNZrW%2B5O904D3y7JShxWwED5LgSU1APwHApRBqlbYoWGoptz6%2BfIA0Ds8oUdd1BYKyf1Y8IIUsbap84AhK9WoEbbFtNVAAiPpbg3FTGPqDXvS8jIXnbyJuvpTIulppMHE56N4m%2FnzFmY%2FZmO6kBx47qXAVkHtYPjDzXZ8p0B%2BrgI9WZq%2FfZOGAOq4VcuXxmEuS0yu52Bn8m2ImSyE02bPeSLDBzEcBRfGEdhE%2Ff1293GGIvUhUvSqU%2B6E6Q3oGsuuU1v9l%2F9cXc95fZdd%2FBXuzYqCSN7ucHGvlnbTNNbLjrzzW3cBY1bLiZmFanxHTitffCbZupEbtYJQC8SMKyYsBop0T6k4NcsMRGByAgZKqdbhLOrKMezveQO3xsurXwTzbvVnN%2FKUGySWcFhP5fwkhuhFYMItiYO8MfXfKa57Pdai3%2BM0DoNEb80HfZEhL5%2B8qdSAFCApFvtXwcWswb4IsDDJkrK%2BBjqwAR50inE6zABnUAMaleyf2wNh4E2Ad6PKC8xMXekh3yB8E8jnsSYfnsIJXOfo6duyAupaIjmHigDtYafRb47S%2B6Y4kzbtMjcahW7AV%2BGQ13dczT8py2n705y2TYD8U8e0PU9%2Bj9tHqEil9f5vgzpL3aY9LvDhKWGcLpQwOZQE%2FFU%2B%2FNchhzCKM%2BH0s%2F5t66fbuBgvgW9o5WNW4raRHLgB1SoxmE3OnTm0mQdnnfAEKEX%2B&Expires=1741587663\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.242.249, 52.92.191.153, 52.92.241.169, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.242.249|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1342334 (1.3M) [binary/octet-stream]\n",
            "Saving to: â€˜ENCFF367KIF.bed.gzâ€™\n",
            "\n",
            "ENCFF367KIF.bed.gz  100%[===================>]   1.28M  1.69MB/s    in 0.8s    \n",
            "\n",
            "2025-03-08 18:21:04 (1.69 MB/s) - â€˜ENCFF367KIF.bed.gzâ€™ saved [1342334/1342334]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "4K71Ag8KMVSL"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}